common/playlist.c

```diff
@@ -147,7 +148,7 @@ void playlist_shuffle(struct playlist *pl)
     for (int n = 0; n < pl->num_entries; n++)
         pl->entries[n]->original_index = n;
     for (int n = 0; n < pl->num_entries - 1; n++) {
-        int j = (int)((double)(pl->num_entries - n) * rand() / (RAND_MAX + 1.0));
+        size_t j = mp_rand_next() % (pl->num_entries - n);
```

Comment:
This change seems unjustified and questionable

Reply:
There's no RAND_MAX for this, it's UINT64_MAX, wasn't sure if casting it to double was really the right call. UNIT64_MAX is so large that just taking the modulus is just much simpler.

Reply:
Yeah, that's a fair point. I was also worrying about whether or not these cases are evenly spaced, but actually, I think the old double code suffers the same flaw.